pkg/export/export.go

```diff
@@ -262,7 +310,7 @@ func (alwaysLease) OnLeaderChange(func()) {
 	// We never lose the lease as it's always owned.
 }
 
-func newMetricClient(ctx context.Context, opts ExporterOpts) (*monitoring.MetricClient, error) {
+func defaultNewMetricClient(ctx context.Context, opts ExporterOpts) (metricServiceClient, error) {
```

Comment:
question: Do we need to support _every_ command line argument for hot-reloading? IIUC we only need to support options configurable in `OperatorConfig.Collection`, which would be limited to:
- export.match
- export.credentials

after #965 is merged.

I was hoping we could avoid recreating the metric client, but I don't think we can if we want to support resetting `export.credentials`..